tutorials/working_with_auxiliary_data.md

```diff
@@ -0,0 +1,89 @@
+# Working with auxiliary data
+
+This tutorial gives an overview of how to deal with auxiliary information in recommender systems using Cornac. In our context, auxiliary data stands for information beyond user-item interactions or preferences, which often hold a clue on how users consume items. Examples of such information or modalities are: item textual descriptions, user/item reviews, product images, social network, etc.
+
+## Modality classes and utilities
+
+In addition to implementing readers and utilities for different types of data, the `cornac.data` module provides modality classes, namely GraphModality, ImageModality and TextModality. The purpose of the latter classes is to make it convinient to work with the corresponding modalities by:
+
+- Offering a number of useful routines for data formatting, representation, manipulation and transformation.
+- Freeing users from the tedious process of aligning auxiliary data with the set of training users, items or ratings.
+- Enabling cross-utilization of models designed for one modality to work with a different modality. This topic is covered by the tutorials under the [Cross-Modality](./README.md#Cross-Modality) section.    
+
+In the following we will discover the text modality by going through a concrete example involving text auxiliary information. The same principles would apply for the other modalities, when dealing with graph (e.g, social network) or visual (e.g., product images) auxiliary data.
```

Comment:
In the following`,` we will discover the text modality by going through a concrete example involving text auxiliary information. The same principles would apply for the other modalities when dealing with graph data (e.g, social network) or visual data (e.g., product images).